samples/synchronization: Fix outdated comments and code structure

xvigo opened the pull request:
This commit has no functional changes. It makes the comments up to date with the code and changes the code structure to ensure consistency. 

These inconsistencies were introduced when the static thread definition was replaced with dynamic, but the comments remained unchanged.

 - Fixed outdated comments about the static approach that were not corresponding to the code.
 - Rearrange thread definitions, to obtain similar code structure for both threads.

kartben commented on samples/synchronization/src/main.c:
```diff
@@ -10,10 +10,10 @@
  * to take turns printing a greeting message at a controlled rate. The demo
- * shows both the static and dynamic approaches for spawning a thread; a real
- * world application would likely use the static approach for both threads.
+ * shows only the dynamic approach for spawning a thread. Alternatively, 
```
I know this is confusing/awkward but could you please remove the trailing whitespace there? :)

kartben commented on samples/synchronization/src/main.c:
```diff
@@ -10,10 +10,10 @@
- * shows both the static and dynamic approaches for spawning a thread; a real
- * world application would likely use the static approach for both threads.
+ * shows only the dynamic approach for spawning a thread. Alternatively, 
+ * a thread can be declared at compile time by calling K_THREAD_DEFINE.
```
thanks for catching the inaccuracy in the documentation! I think this would make for a great follow-up commit to actually make thread A or B a statically defined one, if possible? :) FWIW looks like the change dates back from 686b9fff5eb3391536185275fb7ac7794ef00dff.

kartben commented on samples/synchronization/src/main.c:
```diff
@@ -68,41 +68,38 @@ void helloLoop(const char *my_name,
+K_THREAD_STACK_DEFINE(threadA_stack_area, STACKSIZE);
+static struct k_thread threadA_data;
+
+
```
remove extra line

kartben commented:
Thanks for the PR @xvigo, and sorry for the delay! Please check my comments

kartben commented:
Hi @xvigo -- when you have a chance, could you look at addressing the review comments and compliance check errors? Thanks in advance!

kartben commented:
@xvigo ping?

kartben commented:
@xvigo I would love to help you get this fixed - let me know if you need any assistance

xvigo committed:
samples: synchronization: fix comments & structure

This commit has no functional changes. It makes the comments up to date
with the code and changes the code structure to ensure consistency.

These inconsistencies were introduced when the static thread definition
was replaced with dynamic, but the comments remained unchanged.

 - Fixed outdated comments about the static approach that were
   not corresponding to the code.
 - Rearrange thread definitions, to obtain similar code structure
   for both threads.

Signed-off-by: Vilem Gottwald <xvigo.dev@gmail.com>

xvigo committed:
samples: synchronization: add static thread

Replace one of the dynamic threads with a static thread to show the
all the possible ways of creating threads.

Signed-off-by: Vilem Gottwald <xvigo.dev@gmail.com>

kartben commented:
hey @xvigo, thank you for investing time in cleaning this up and making sure the diff looks nice, much appreciated! Looks like you will need to properly sign off the commit if you have a chance. Thanks again.

xvigo commented:
@kartben Should be ready. Sorry for the delay.

kartben commented:
@xvigo almost there! :) the email is the sign off entry should be between angle brackets. Thanks!

xvigo commented:
I see, I just replaced the GitHub noreply email with this new private one and didn't really think about it, what a dumb mistake. 
Sorry for the delays, I never actually used Zephyr and didn't have much time lately, so it wasn't my priority. 
Hopefully, it will pass now :)

kartben reviewed (approved):
Awesome, thanks a bunch, @xvigo!

@nashif that sample was pretty stale and this is a great and much needed refresh, please have a look when you have a chance.
